src/js/components/projects/createForm.tsx

```diff
@@ -69,15 +90,181 @@ const ProjectForm = ({
+    /* istanbul ignore else */
+    if (isMounted.current) {
+      setRepoBranches(baseBranches);
+      setFetchingBranches(false);
+    }
+  };
+
+  const setBranch = (branch: string) => {
+    setInputs({ ...inputs, branch_name: branch });
+  };
+
+  const resetBranchForm = () => {
+    setBranch('');
+    resetFilterVal();
+  };
+
+  const closeBranchForm = () => {
+    resetBranchForm();
+    setFromBranchChecked(false);
+  };
+
+  const handleBranchCheckboxChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    if (event.target.value === 'existing') {
+      doGetBranches();
+      setFromBranchChecked(true);
+    } else {
+      closeBranchForm();
+    }
+  };
+
+  const handleBranchSelection = (
+    event: any,
+    { selection }: { selection: ComboboxOption[] },
+  ) => {
+    setBranch(selection[0].label);
+  };
+
+  const handleBranchChange = (event: any, { value }: { value: string }) => {
+    setFilterVal(value);
+  };
+
+  const handleBranchRemoveSelection = () => {
+    resetBranchForm();
+  };
+
+  const handleBranchBlur = () => {
+    if (repoBranches.includes(filterVal)) {
+      setBranch(filterVal);
+    }
+    resetFilterVal();
+  };
+
+  const noFeatureBranches = !repoBranches.length;
+  const inputVal = inputs.branch_name;
+  const branchOptions: ComboboxOption[] = repoBranches.map((item, index) => ({
+    id: `${index + 1}`,
+    label: item,
+  }));
+  const selection = branchOptions.find((branch) => branch.label === inputVal);
+  let noOptionsFoundText = null;
+  if (fetchingBranches) {
+    noOptionsFoundText = i18n.t('Loading existing branches…');
+  } else {
+    noOptionsFoundText = (
+      <>
+        {noFeatureBranches
+          ? i18n.t("There aren't any available branches at this time.")
+          : i18n.t('No matching branches found.')}{' '}
+        <Button
+          label={i18n.t('Check again')}
+          variant="link"
+          onClick={doGetBranches}
+        />{' '}
+        {i18n.t('to refresh this list or')}{' '}
+        <Button
+          label={i18n.t('start from a new branch.')}
+          variant="link"
+          onClick={closeBranchForm}
+        />
+      </>
+    );
+  }
+
   return (
-    <form onSubmit={handleSubmit} className="slds-form slds-m-bottom--large">
+    <form
+      onSubmit={handleSubmit}
+      className="slds-form slds-m-bottom--large"
+      data-form="create-project-branch"
+    >
       {isOpen && (
         <>
+          <p className="slds-form-element__help slds-m-bottom_small">
+            <Trans i18nKey="createProjectHelpText">
+              Projects in Metecho are the high-level features that can be broken
+              down into smaller parts by creating Tasks. You can create a new
+              project or create a project based on an existing GitHub branch.
+              Every project requires a unique project name, which becomes the
+              branch name in GitHub unless you choose to use an existing branch.
+            </Trans>
```

Comment:
Is it worthwhile to try showing this text initially, and subsequently and as a tool-tip? (Mentioning as it's in the notes in Trello)